js/adapt-languagePicker.js

```diff
@@ -1,44 +1,46 @@
-define([
-  'core/js/adapt',
-  './languagePickerView',
-  './languagePickerNavView',
-  './languagePickerModel'
-], function(Adapt, LanguagePickerView, LanguagePickerNavView, LanguagePickerModel) {
+import Adapt from 'core/js/adapt';
+import LanguagePickerView from './languagePickerView';
+import LanguagePickerNavView from './languagePickerNavView';
+import LanguagePickerModel from './languagePickerModel';
 
-  var languagePickerModel;
+let languagePickerModel;
 
-  Adapt.once('configModel:dataLoaded', onConfigLoaded);
-
-  /**
+class LanguagePicker extends Backbone.Controller {
+  
+  initialize() {
+    this.listenTo(Adapt, 'configModel:dataLoaded', this.onConfigLoaded);
+  }
+  
+    /**
    * Once the Adapt config has loaded, check to see if the language picker is enabled. If it is:
    * - stop the rest of the .json from loading
    * - set up the language picker model
    * - register for events to allow us to display the language picker icon in the navbar on pages and menus
    * - wait for offline storage to be ready so that we can check to see if there's a stored language choice or not
    */
-  function onConfigLoaded() {
-    if (!Adapt.config.has('_languagePicker')) return;
-    if (!Adapt.config.get('_languagePicker')._isEnabled) return;
+  onConfigLoaded() {
+    const config = Adapt.config.get('_languagePicker');
+    if (!config?._isEnabled) return;
 
     Adapt.config.set('_canLoadData', false);
 
-    languagePickerModel = new LanguagePickerModel(Adapt.config.get('_languagePicker'));
+    languagePickerModel = new LanguagePickerModel(config);
 
-    Adapt.on('router:menu router:page', setupNavigationView);
+    Adapt.on('router:menu router:page', this.setupNavigationView);
```

Comment:
Here and on line 35, the controller can do its own listening, i.e. `this.listenTo()`/`this.listenToOnce()`.